includes/html/pages/device/routing/isis.inc.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use App\Models\IsisAdjacency;
+use App\Models\Port;
+
+echo '
+<div>
+  <div class="panel panel-default">
+    <div class="panel-body">
+      <table class="table table-condensed table-hover" style="border-collapse:collapse;">
+        <thead>
+          <tr>
+            <th>&nbsp;</th>
+            <th>Local device</th>
+            <th>Local interface</th>
+            <th>Adjacent</th>
+            <th>System ID</th>
+            <th>Area</th>
+            <th>System type</th>
+            <th>State</th>
+            <th>Last uptime</th>
+          </tr>
+        </thead>';
+
+foreach (IsisAdjacency::where('device_id', $device['device_id'])->get() as $adj) {
+    if ($adj->isisISAdjState == 'up') {
+        $color = 'green';
+    } else {
+        $color = 'red';
+    }
+
+    $interface_name = Port::query()->where('port_id', $adj->port_id)->first()->ifName;
```

Comment:
creating N+1 queries, use `with()` on your IsisAdjacency query.

Reply:
`IsisAdjacency::where('device_id', $device['device_id'])->with('port')->get()`

and then `$adj->port->ifName`

Reply:
Fixed